src/Microsoft.NetFramework.Analyzers/Core/SerializationRulesDiagnosticAnalyzer.cs

```diff
@@ -178,12 +181,11 @@ public void AnalyzeSymbol(SymbolAnalysisContext context)
                     {
                         // Look for a serialization constructor.
                         // A serialization constructor takes two params of type SerializationInfo and StreamingContext.
-                        IMethodSymbol serializationCtor = namedTypeSymbol.Constructors.Where(c => c.Parameters.Count() == 2 &&
-                                                                                        c.Parameters[0].Type ==
-                                                                                        _serializationInfoTypeSymbol &&
-                                                                                        c.Parameters[1].Type ==
-                                                                                        _streamingContextTypeSymbol)
-                            .SingleOrDefault();
+                        IMethodSymbol serializationCtor = namedTypeSymbol.Constructors
+                            .SingleOrDefault(
+                                c => c.Parameters.Length == 2 &&
+                                     c.Parameters[0].Type == _serializationInfoTypeSymbol &&
```

Comment:
Unrelated to your refactoring, but we should use `.Equals` instead of equals operators for symbol equality checks. Don't need to address this in this PR.